### ISSUE TITLE
Parcel 2 generates wrong declaration file when `declare global` is used

### ISSUE THREAD

**Issue by jakubmazanec** [parcel-bundler/parcel#7455]
`declare global` in TypeScript file makes Parcel generate wrong declaration file:

```TypeScript
// src/index.ts
declare global {}

class Foo {
}

export const foo = new Foo();
```

```TypeScript
// dist src/index.d.ts
// no foos? 😭
```

## 🎛 Configuration

Simple setup with config only via package.json

```json
{
  "name": "parcel-bug-declare-global",
  "version": "1.0.0",
  "main": "dist/index.js",
  "source": "src/index.ts",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "parcel build"
  },
  "devDependencies": {
    "@parcel/packager-ts": "^2.0.1",
    "@parcel/transformer-typescript-types": "^2.0.1",
    "parcel": "^2.0.1",
    "typescript": "4.4.4"
  }
}
```

## 🤔 Expected Behavior

After running `npm install` and `npm build`, file `dist/index.d.ts` should look maybe something like this?

```TypeScript
declare global { }
declare class Foo {
}
export const foo: Foo;
```

## 😯 Current Behavior

Ouput definition file is completely empty.

## 💻 Code Sample

[Sample repo](https://github.com/jakubmazanec/parcel-bug-declare-global)

**Comment by Shinyaigeek** [comment]
This issue seems to be fixed in #7315 👀  current `v2` branch correctly works.

**Comment by mischnic** [comment]
Indeed, it also works fine with the nightly versions.
